lib/kubebox.js

```diff
@@ -153,6 +153,86 @@ class Kubebox extends EventEmitter {
       carousel.move();
     });
 
+    dashboard.on('exec', (namespace, pod, container) => {
+      // check if connection already exists
+      const user = kube_config.current_context.user.username;
+      const id = namespace + pod + container;
+      function wsexecExists (_function) {
+        if (!_function.id)
+          return false;
+        return _function.id === id;
+      }
+      const index = carousel.pages.findIndex((wsexecExists));
+      if (index >= 0) {
+        carousel.currPage = index;
+        carousel.move();
+        return;
+      }
+      
+      // connect
+      const exec = new Exec(screen, namespace, pod, container, user);
+      const { promise, cancellation, sendData, getSocket} = get(client.exec(namespace, pod, container, exec.termName()), { stream: () => exec.print() });
```

Comment:
I'd be inclined to put the following logic into the `exec` widget for encapsulation and reuse.

Reply:
Add a spinner.

Reply:
Let me work on this.